doc/kci_testsuite.md

```diff
@@ -0,0 +1,300 @@
+Upload rootfs image to kernelci-backend (storage)
+
+```
+kci_rootfs upload --rootfs-dir /path/to/rootf/image/ --upload-path server/path --api http://api.kernelci --token KCI_API_TOKEN
+```
+
+Add rootfs definition to `test-configs.yaml`
+
+```yaml
+  debian_buster-v4l2_ramdisk:
+    type: debian
+    ramdisk: 'buster-v4l2/20200626.0/{arch}/rootfs.cpio.gz'
+```
+
+`ramdisk` path must be relative to storage server URL.
+
+Detailed description of the test configuration can be found [here](https://github.com/kernelci/kernelci-doc/wiki/Test-configurations)
+
+2. Add LAVA job definition template
+
+Create appropriate directory structure under `templates` directory.
+
+```
+mkdir templates/v4l2-compliance/
+```
+
+Create test plan definition
+
+```
+cat templates/v4l2-compliance/v4l2-compliance.jinja2
+```
+
+```yaml
+- test:
+    timeout:
+      minutes: 10
+    definitions:
+    - repository:
+        metadata:
+          format: Lava-Test Test Definition 1.0
+          name: {{ plan }}
+          description: "v4l2 test plan"
+          os:
+          - debian
+          scope:
+          - functional
+        run:
+          steps:
+          - /usr/bin/v4l2-parser.sh {{ v4l2_driver }}
+      from: inline
+      name: {{ plan }}
+      path: inline/{{ plan }}.yaml
+```
+
+
+```
+cat templates/v4l2-compliance/generic-qemu-v4l2-compliance-template.jinja2 
+```
+
+```
+{% extends 'boot/generic-qemu-boot-template.jinja2' %}
+{% block actions %}
+{{ super () }}
+
+{% include 'v4l2-compliance/v4l2-compliance.jinja2' %}
+
+{% endblock %}
+
+{%- block image_arg %}
+        image_arg: '-kernel {kernel} -append "console={{ console_dev }},115200 root=/dev/ram0 debug verbose {{ extra_kernel_args }}"'
+{%- endblock %}
+```
+
+3. Update `test-plans` section of `test-configs.yaml` with new test definition
+
+
+```yaml
+  v4l2-compliance-vivid:
+    rootfs: debian_buster-v4l2_ramdisk
+    pattern: 'v4l2-compliance/generic-qemu-v4l2-compliance-template.jinja2'
+    params: {v4l2_driver: "vivid"}
+    filters:
+      - whitelist:
+          defconfig:
+            - 'defconfig+virtualvideo'
+            - 'multi_v7_defconfig+virtualvideo'
+```
+
+4. Use `kci_build` to build the kernel and `kci_test` to run the tests in a LAVA lab.
+
+Build the Kernel
+
+```
+kci_build build_kernel --kdir /path/to/linux --defconfig x86_64_defconfig+virtualvideo --arch x86_64 --build-env gcc-8 --verbose -j 4
+```
+
+Genrate test job definitions
+
+```
+kci_test generate --bmeta-json /path/to/bmeta.json --dtbs-json /path/to/artifacts/dtbs.json --lab-json /path/to/mgalka-lava-local.json --storage http://storage.kernelci --lab your-lab-name --user lava_user --token LAVA_API_TOKEN --output /output/path --callback-id your-lava-callback-name --callback-url http://api.kernelci
+```
+
+Submit test jobs
+
+```
+kci_test submit --lab your-lab-name --user lava_user --token LAVA_API_TOKEN --jobs /path/to/generated/test/job/definitions
+```
+
+5. Create PR to add the test plan
```

Comment:
Please add a paragraph here to explain what this is about, to send the test plan upstream in KernelCI and have it used on kernelci.org.  It's related to the comment I put earlier about creating new rootfs images, with a PR this can be done on staging.  Also, some people will be using their own instance of KernelCI and won't need to make a PR.